zerver/models/prereg_users.py

```diff
@@ -49,6 +49,14 @@ class PreregistrationRealm(models.Model):
         UserProfile, null=True, related_name="+", on_delete=models.SET_NULL
     )
 
+    IMPORT_FROM_CHOICES = {
+        "None": 100,
+        "Slack": 200,
+    }
+    import_from = models.PositiveSmallIntegerField(default=IMPORT_FROM_CHOICES["None"])
+    slack_access_token = models.CharField(max_length=100, default="")
+    uploaded_import_file_name = models.CharField(max_length=255, default="")
```

Comment:
We don't generally use JSON fields, but I think this is probably a case where an `data_import_metadata` JSON field might be more appropriate, since it'll be custom fields for each type of import. See `extra_data` on RealmAuditLog for an example.

Reply:
Okay, changed it to use this:

```python
    IMPORT_FROM_CHOICES = [
        ("none", "Don't import"),
        ("slack", "Import from Slack"),
    ]
    data_import_metadata = models.JSONField(default=dict, encoder=DjangoJSONEncoder)
```

Is that what you had in mind?